src/app/time-select/time-select.component.ts

```diff
@@ -0,0 +1,189 @@
+import {ECalendarValue} from '../common/types/calendar-value-enum';
+import {SingleCalendarValue} from '../common/types/single-calendar-value';
+import {ECalendarType} from '../common/types/calendar-type-enum';
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+  SimpleChanges,
+  forwardRef,
+  HostBinding
+} from '@angular/core';
+import {TimeSelectService, TimeUnit} from './time-select.service';
+import * as moment from 'moment';
+import {Moment} from 'moment';
+import {ITimeSelectConfig} from './time-select-config.model';
+import {
+  NG_VALUE_ACCESSOR,
+  NG_VALIDATORS,
+  ControlValueAccessor,
+  Validator,
+  ValidationErrors,
+  FormControl
+} from '@angular/forms';
+import {CalendarValue} from '../common/types/calendar-value';
+import {UtilsService} from '../common/services/utils/utils.service';
+import {IMonthCalendarConfig} from '../month-calendar/month-calendar-config';
+import {IMonth} from '../month-calendar/month.model';
+
+@Component({
+  selector: 'dp-time-select',
+  templateUrl: 'time-select.component.html',
+  styleUrls: ['time-select.component.less'],
+  providers: [
+    TimeSelectService,
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => TimeSelectComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => TimeSelectComponent),
+      multi: true
+    }
+  ]
+})
+export class TimeSelectComponent implements OnInit, OnChanges, ControlValueAccessor, Validator {
+
+  @Input() config: ITimeSelectConfig;
+  @Input() displayDate: SingleCalendarValue;
+  @Input() minDate: Moment;
```

Comment:
we should also support strings, that means that this component should also receive a format as an input.